modules/drivers/mongo/test/metabase/driver/mongo/execute_test.clj

```diff
@@ -53,3 +67,69 @@
                            [1 "Rasta"     now nil]]
                     :columns ["_id" "name" "last_login" "alias"]}
                    (mt/rows+column-names (qp/process-query query))))))))))
+
+(defn- interrupt-ex? [^MongoCommandException ex]
+  (and (= 11601 (.getErrorCode ex))
+       (= "Interrupted" (.getErrorCodeName ex))))
+
+(deftest kill-an-in-flight-query-method-validation-test
+  (testing "Confirm that method used to kill in-flight queries on mongo works"
+    (mt/test-driver
+     :mongo
+     (mt/dataset
+      sample-dataset
+      (let [;; Following query runs on my m2 ~ 50 seconds.
+            query (mt/mbql-query orders
+                                 {:aggregation [[:sum $total]],
+                                  :breakout [!month.created_at],
+                                  :order-by [[:asc !month.created_at]],
+                                  :joins [{:alias "People_User",
+                                           :strategy :left-join,
+                                           :condition
+                                           [:!= $user_id &People_User.people.id],
+                                           :source-table $$people}]})
+            compiled (-> query qp/compile :query)
+            pipeline (m.util/into-array-list (m.conversion/to-db-object compiled))]
+        (mongo.util/with-mongo-connection [connection (mt/id)]
+          (let [client-database ^MongoDatabase (#'mongo.execute/connection->database connection)
+                collection ^MongoCollection (. client-database (getCollection "orders"))]
+            (with-open [session ^ClientSession (#'mongo.execute/start-session! connection)]
+              (let [aggregate (.aggregate collection session ^java.util.ArrayList pipeline)
+                    ;; Manually tested: if session is closed before aggregation execution takes place (call to
+                    ;; eg. either `.into` or `.cursor`) aggregation is then executed.
+                    ;; TODO: Find workaround!
```

Comment:
Does `.into` leave its mark somewhere on the observable world? If yes, we should look for that mark. If not, then we can only hope that we wait enough before we kill the session. In this latter case, we could set a semaphore in the `a/thread` just before calling `.into` and wait on it before starting the `Thread/sleep`. That would reduce the chance of killing the session ahead of time to a minimum.

Reply:
Test was removed as advised later.